parsec/core/gui/custom_dialogs.py

```diff
@@ -324,23 +335,67 @@ def _exec_method(
                 pool.close()
                 cls.pools.pop(target_cls, None)
 
-    getOpenFileName = classmethod(
-        generate_dialog_method(QFileDialog, QFileDialog.getOpenFileName, ("", ""))
-    )
-    getOpenFileNames = classmethod(
-        generate_dialog_method(QFileDialog, QFileDialog.getOpenFileNames, (cast(list[str], []), ""))
-    )
-    getExistingDirectory = classmethod(
-        generate_dialog_method(QFileDialog, QFileDialog.getExistingDirectory, "")
-    )
-    getSaveFileName = classmethod(
-        generate_dialog_method(QFileDialog, QFileDialog.getSaveFileName, ("", ""))
-    )
-    print_html = classmethod(
-        generate_dialog_method(
-            _subprocess_dialog.PrintHelper, _subprocess_dialog.PrintHelper.print_html, None
-        )
-    )
+    # getOpenFileName = classmethod(
+    #     generate_dialog_method(QFileDialog, QFileDialog.getOpenFileName, ("", ""))
+    # )
+    @classmethod
+    @copy_args_types(QFileDialog.getOpenFileName)
+    def getOpenFileName(cls, *args: Any, **kwargs: Any) -> tuple[str, str]:
+        try:
+            return cls._exec_method(QFileDialog, QFileDialog.getOpenFileName, *args, **kwargs)
+        except cls.DialogCancelled:
+            return ("", "")
```

Comment:
I needed to change how those `classmethod` where generated because I was getting the error

```txt
Argument 1 to "__get__" of "classmethod" has incompatible type "None"
```
From `mypy`, when we where calling those methods like: 

https://github.com/Scille/parsec-cloud/blob/0d56def78ed028a0f44024454fc67adabf7420a8/parsec/core/gui/files_widget.py#L829-L831

I couldn't find why I was getting that error

Maybe @vxgmichel you have a idea on that ?